frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/virtual-hardware-tab/virtual-hardware-tab.tsx

```diff
@@ -0,0 +1,263 @@
+          source = 'Windows Tools';
+        }
+      }
+
+      if (source === StorageUISource.IMPORT_DISK.getValue()) {
+        name = volumeWrapper.getPersistentVolumeClaimName();
+      }
+
+      if (source === StorageUISource.URL.getValue()) {
+        name = dataVolumeWrapper.getURL();
+      }
+
+      return {
+        wizardStorageData,
+        source,
+        isCD: true,
+        name,
+        storageClass: combinedDisk.getStorageClassName(),
+      };
+    });
+};
+
+const VirtualHardwareTabFirehose: React.FC<VirtualHardwareTabFirehoseProps> = ({
+  wizardReduxID,
+  isLocked,
+  setTabLocked,
+  removeStorage,
+  storages,
+  persistentVolumeClaims,
+}) => {
+  const virtualStorages = getVirtualStoragesData(storages, persistentVolumeClaims);
+  const showStorages = virtualStorages.length > 0;
+  const disableAddCD = virtualStorages.length > 1;
+  const availableCDName = getAvailableCDName(storages.map((storage) => storage.disk));
+  const withProgress = wrapWithProgress(setTabLocked);
+
+  const addButtonProps = {
+    id: 'attach-cdrom',
+    className: 'pf-m-link--align-left',
+    onClick: () =>
+      withProgress(
+        vmWizardStorageModalEnhanced({
+          initialType: DiskType.CDROM,
+          blocking: true,
+          wizardReduxID,
+          availableCDName,
+        }).result,
+      ),
+    isDisabled: isLocked || disableAddCD,
+  };
+
+  return (
+    <div className="kubevirt-create-vm-modal__virtual-hardware-tab-container">
+      <Split>
+        <SplitItem isFilled>
+          <Title headingLevel="h5" size="lg">
+            Virtual Hardware
+          </Title>
+        </SplitItem>
+      </Split>
+      <Split className="kubevirt-create-vm-modal__virtual-hardware-tab-cd-title">
+        <SplitItem isFilled>
+          <Title headingLevel="h4" size="md">
+            CD-ROMs
+          </Title>
+        </SplitItem>
+      </Split>
+      {showStorages && (
+        <>
+          <div className="kubevirt-create-vm-modal__virtual-hardware-tab-main">
+            <VMDisksTable
+              columns={[
+                {
+                  title: 'Content',
+                  sortField: 'name',
+                  transforms: [sortable],
+                },
+                {
+                  title: 'Source',
+                  sortField: 'source',
+                  transforms: [sortable],
+                },
+                {
+                  title: 'Storage Class',
+                  sortField: 'storageClass',
+                  transforms: [sortable],
+                },
+                {
+                  title: '',
+                },
+              ]}
+              columnClasses={cdTableColumnClasses}
+              data={virtualStorages}
+              customData={{ isDisabled: isLocked, withProgress, removeStorage, wizardReduxID }}
+              row={VmWizardStorageRow}
+            />
+
+            <SplitItem>
+              <div className="kubevirt-add-cd-btn">
```

Comment:
What is this div being used for?

Reply:
the add cd button, the classname has no effect atm

Reply:
right so it can be removed?